### PR TITLE
make nodepool AZs optional

### PR DESCRIPTION
**Pull request by anvddriesch**
Towards: https://github.com/giantswarm/giantswarm/issues/14429

We want to make the nodepool AZs optional so they can be defaulted. 

## Checklist

- [ ] Consider SIG UX feedback.
- [ ] Update changelog in CHANGELOG.md.
- [ ] If adding a new type, ensure that it is added to the [CRD unit tests](https://github.com/giantswarm/apiextensions/blob/d78aba91d578d56ef49f58e336035d35edc4e1b0/pkg/crd/crd_test.go#L9).

### CODE REVIEW

**Review by njuettner — APPROVED** [review]
lgtm

**Comment by kopiczko** [comment]
Changelog is missing.

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Changed
+
+- Make `availabilityZones` attribute in `AWSMachineDeployment` optional.
+
```
Comment (marians) [inline CHANGELOG.md:15]: Would be great to add information regarding the default. As a user, when I don't set this attribute, what will happen?

Comment (anvddriesch) [reply CHANGELOG.md:15]: Good point! :) This will be defaulted by the `aws-admission-controller` if it is not set.

**Pull request merged**